test/helpers.py

```diff
@@ -119,7 +119,7 @@ def build_goaway_frame(self,
         f = GoAwayFrame(0)
         f.error_code = error_code
         f.last_stream_id = last_stream_id
-        f.additional_data = additional_data
+        f.additional_data = additional_data or b''
```

Comment:
Ah, I see what's happened here. It might be better, rather than adding this fallback, to parametrize two arguments: the input value, and the expected result. That makes the test more explicit about what's going on, rather than hiding this in the frame factory.